moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1702,7 +1703,15 @@ bool PlanningScene::processCollisionObjectMsg(const moveit_msgs::CollisionObject
       if (s)
       {
         Eigen::Affine3d p;
-        tf::poseMsgToEigen(object.primitive_poses[i], p);
+        // To normalize any input quaternion
+        geometry_msgs::Pose primitive_pose = object.primitive_poses[i];
+        tf::Quaternion tf_quaternion;
+        tf::quaternionMsgToTF(object.primitive_poses[i].orientation, tf_quaternion);
+        tf_quaternion.normalize();
+        geometry_msgs::Quaternion msg_quaternion;
+        quaternionTFToMsg(tf_quaternion, msg_quaternion);
+        primitive_pose.orientation = msg_quaternion;
+        tf::poseMsgToEigen(primitive_pose, p);
```

Comment:
This is quite a bulk of code and I am pretty sure this is not the only place in this file where quaternions *should* be normalized.
Can you please look through the rest of the places that use quaternions here and wrap the validation into a function?